### PR TITLE
feat(memphis.ts): Added a logger and option to suppress logs

### PR DESCRIPTION
**Pull request by origranot**
Hey,

While developing with Memphis I noticed there are logs being printed to the console every time the status of the broker changes.

I think it is a good idea to add a way to suppress this logs / delete them (I don't think this logs should be printed in production - at least not a normal console log without any way to suppress / include timestamps).

In addition I think it might be a good idea to add a logger to Memphis package for better logging and error handling.
Let me know if there is something I should change 💯

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
         "@nestjs/core": "^9.0.0",
         "@nestjs/microservices": "^9.1.4",
         "@nestjs/platform-express": "^9.0.0",
+        "@origranot/ts-logger": "^1.12.0",
```
Comment (idanasulinmemphis) [inline package.json:31]: We want to keep the client as lightweight as possible so we avoid from adding packages we can implement by ourselves.

File: src/memphis.ts [hunk 2]
```diff
@@ -247,14 +253,14 @@ class Memphis {
           for await (const s of this.brokerManager.status()) {
             switch (s.type) {
               case 'update':
-                console.log(`reconnected to memphis successfully`);
+                this.logger.info(`reconnected to memphis successfully`);
```
Comment (idanasulinmemphis) [inline src/memphis.ts:256]: Instead using the new package, please use if statements based on the new flag you have added

**Review by idanasulinmemphis — COMMENTED** [review]
Hi @origranot, thanks for this PR, absolutely needed.
I've added a few comments, please fix them so we will be able to merge

File: src/memphis.ts [hunk 3]
```diff
@@ -153,7 +156,8 @@ class Memphis {
     timeoutMs = 2000,
     keyFile = '',
     certFile = '',
-    caFile = ''
+    caFile = '',
+    suppressLogs =false
```
Comment (idanasulinmemphis) [inline src/memphis.ts:160]: missing space

File: src/nest/server.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ export class MemphisServer extends Server implements CustomTransportStrategy {
             this.connection = await memphis.connect(this.options);
             this.createConsumer();
         } catch (err) {
-            console.log(err);
+            this.logger.error(err);
```
Comment (idanasulinmemphis) [inline src/nest/server.ts:20]: I believe that won't work since no logger defined anywhere

File: lib/consumer.js [hunk 5]
```diff
@@ -83,7 +83,7 @@ class Consumer {
         this.eventEmitter.on(event, cb);
     }
     async fetch({ batchSize = 10 }) {
-        var e_1, _a;
+        var _a, e_1, _b, _c;
```
Comment (idanasulinmemphis) [inline lib/consumer.js:86]: please remove the build files from the PR we will create them internally

**Review by idanasulinmemphis — COMMENTED** [review]
@origranot, I'vee added some more comments

**Comment by origranot** [comment]
> @origranot, I'vee added some more comments

@idanasulinmemphis fixed 🚀

**Comment by origranot** [comment]
@idanasulinmemphis https://github.com/memphisdev/memphis.js/pull/166

**Pull request closed without merging**